File: micropython/prime.py
import math, time
last = 10000
found = 4          # we start from 11, know 2, 3, 5, 7
print('Prime numbers to {}'.format(last))
start = time.ticks_us()
for number in range(11, last, 2):
    prime = True
    for divider in range(3, int(math.sqrt(number))+1, 2):
        if number % divider == 0:
            prime = False
            break
    if prime:
        found += 1
        prime = True
end = time.ticks_us()
print("This took: {} seconds.".format((end - start)/1000000))
print("I found {:.4} prime numbers.".format(found))

File: java/v0.8.1989/Prime.java
class prime {
  public static void main(String[] args) {
    int last = 100000;
    int found = 4;
    System.out.println("Prime numbers to " + last + " in Java v0.8.1989");
    long start = System.nanoTime();
    for(int number = 11; number < last; number = number + 1) {
      boolean prime = true;
      for(int divider = 2; divider < number + 1; divider = divider + 1) {
        if((float)number/divider == (int)number/divider) {
          prime = false;
          break;
        }
      }
      if(prime) {
        found++;
        // System.out.print(", " + number);
      }
    }
    long end = System.nanoTime();
    System.out.print("I found " + found + " prime numbers. ");
    System.out.print("This took me " + ((end - start) / 1000000000.0) + " seconds.");
  }
}
